test/unit/api.test.js

```diff
@@ -0,0 +1,301 @@
+            let actualReturnValue;
+            let callback;
+            let stubbedMethod;
+
+            before('setup error stub for callback case', function () {
+              superPromise = Promise.reject(new Error('error!'));
+              stubbedMethod = makeStub(superPromise);
+              callback = sinon.spy();
+              args[callbackPosition] = callback;
+              actualReturnValue = instance[method](...args);
+            });
+
+            beforeEach(async function () {
+              await superPromise.catch(error => error);
+            });
+
+            it('should return void', () => expect(actualReturnValue).to.be.undefined);
+
+            it('should call the callback with error', () => {
+              expect(callback).to.have.been.calledOnce;
+              const expectedArgs = callback.args[0];
+              expect(expectedArgs).to.have.nested.property('[0].message', 'error!');
+            });
+
+            it(`should pass only (${argsPassedToDriver.join(', ')}) to the driver api`, () => {
+              expect(stubbedMethod).to.have.been.calledOnceWith(...argsPassedToDriver);
+            });
+          })
+        );
+      }
+
+      const args = Array.from({ length: functionLength - 1 }, (_, i) => i);
+
+      resolveSuite.push(() =>
+        describe(`and ${apiName} is called  with (${args.join(', ')})`, () => {
+          let superPromise;
+          let actualReturnValue;
+          let stubbedMethod;
+
+          before('setup success stub for promise case', function () {
+            superPromise = Promise.resolve({ message: 'success!' });
+            stubbedMethod = makeStub(superPromise);
+            actualReturnValue = instance[method](...args);
+          });
+
+          beforeEach(async function () {
+            await superPromise.catch(error => error);
+          });
+
+          it('should return the same promise the driver returns', () => {
+            expect(actualReturnValue).to.equal(superPromise);
+          });
+
+          it('should return a resolved promise', async () => {
+            const result = await actualReturnValue;
+            expect(result).to.have.property('message', 'success!');
+          });
+
+          it(`should pass only (${args.join(', ')}) to the driver api`, () => {
+            expect(stubbedMethod).to.have.been.calledOnceWithExactly(...args);
+          });
+        })
+      );
+
+      rejectsSuite.push(() =>
+        describe(`and ${apiName} is called  with (${args.join(', ')})`, () => {
+          let superPromise;
+          let actualReturnValue;
+          let stubbedMethod;
+
+          before('setup error stub for promise case', function () {
+            superPromise = Promise.reject(new Error('error!'));
+            stubbedMethod = makeStub(superPromise);
+            actualReturnValue = instance[method](...args);
+          });
+
+          beforeEach(async function () {
+            await superPromise.catch(error => error);
```

Comment:
we don't need to save the error here, but then we need to assign it in the before to its own variable and then check it exactly in the `should return the reject promise` block

Reply:
Moved the awaiting to the test, and we check instance equality now